### PR TITLE
implemented percent based profit requirement

### PR DESCRIPTION
**Pull request by markuspluna**

### CODE REVIEW

File: crates/strategies/blend-strategies/src/auction_manager.rs [hunk 1]
```diff
@@ -351,7 +357,14 @@ fn get_fill_info(min_profit: i128, lot_value: i128, bid_value: i128) -> (i128, i
...
         let profit = mod_lot_value - mod_bid_value;
-        if profit >= min_profit {
+        // check if our profit has reached the minimum profit percentage
+        if profit > 0
+            && profit
+                .fixed_div_floor(mod_bid_value, SCALAR_7)
+                .unwrap_or(1000000)
+                >= min_profit_pct
+        {
+            println!("profit: {:?}, mod bid value: {:?}", profit, mod_bid_value);
```
Comment (mootz12) [inline crates/strategies/blend-strategies/src/auction_manager.rs:367]: We should use the info logger here, or remove the println statement

File: crates/strategies/blend-strategies/src/auction_manager.rs [hunk 2]
```diff
@@ -351,7 +357,14 @@ fn get_fill_info(min_profit: i128, lot_value: i128, bid_value: i128) -> (i128, i
...
         }
         let profit = mod_lot_value - mod_bid_value;
-        if profit >= min_profit {
+        // check if our profit has reached the minimum profit percentage
+        if profit > 0
+            && profit
+                .fixed_div_floor(mod_bid_value, SCALAR_7)
+                .unwrap_or(1000000)
+                >= min_profit_pct
+        {
```
Comment (mootz12) [inline crates/strategies/blend-strategies/src/auction_manager.rs:366]: Assuming the `unwrap_or` was supposed to be 100%?

This being said, the case where `mod_bid_value == 0` (i.e. block 400+, or potentially block 399 with a very small auction), we might be better off checking that case specifically. In the event of any other failure during `fixed_div_floor`, we might be better off not returning from the for loop (I.E. `unwrap_or(0)`)

**Review by mootz12 — CHANGES_REQUESTED** [review]

**Review by mootz12 — APPROVED** [review]
LG2M

File: crates/strategies/blend-strategies/src/auction_manager.rs (hunk 2 above)
Comment (markuspluna) [reply crates/strategies/blend-strategies/src/auction_manager.rs]: yeah that's better

**Review by mootz12 — APPROVED** [review]

**Pull request merged**